local.c

```diff
@@ -493,12 +495,17 @@ static ssize_t local_get_buffer(const struct iio_device *dev,
 }
 
 static ssize_t local_read_dev_attr(const struct iio_device *dev,
-		const char *attr, char *dst, size_t len, enum iio_attr_type type)
+				   unsigned int buf_id, const char *attr,
+				   char *dst, size_t len,
+				   enum iio_attr_type type)
 {
 	FILE *f;
 	char buf[1024];
 	ssize_t ret;
 
+	if (buf_id > 0)
```

Comment:
nit: IMO, `if (!buf_id)` would be more readable. Like this, you are implicitly "playing" with the fact that buf_id is unsigned

Reply:
Writing `if (!buf_id)` (which is the same as `buf_id == 0`) would change the behaviour though.

Reply:
Ups :man_facepalming: . Yeah, different condition...